transaction/std/src/memo/authenticated_sender_with_payment_request_id.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2018-2021 The MobileCoin Foundation
+
+//! Object for 0x0101 Authenticated Sender With Payment Request Id memo type
+
+use super::{
+    authenticated_common::compute_category1_hmac, credential::SenderMemoCredential,
+    RegisteredMemoType,
+};
+use crate::impl_memo_type_conversions;
+use core::convert::TryInto;
+use mc_account_keys::{AddressHash, PublicAddress};
+use mc_crypto_keys::{
+    CompressedRistrettoPublic, KexReusablePrivate, RistrettoPrivate, RistrettoPublic,
+};
+use subtle::{Choice, ConstantTimeEq};
+
+/// A memo that the sender writes to convey their identity in an authenticated
+/// but deniable way, for the recipient of a TxOut, which also includes a
+/// payment request id number under the MAC.
+///
+/// See MCIP document for a discussion of the deniability property.
```

Comment:
Possible to link to the doc?

Reply:
i'm going to follow up on where the mcip draft is